### ISSUE TITLE
Script properties interfer with the node

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#5532]
Tested on Godot 2.0.3 on Windows 10 64 bits

I wrote a simple script to render a laser pointer and put it on a `Sprite` node.
However, as soon as I reloaded the scene, the sprite was unexpectedly offset.
It turns out my `laser_pointer.gd` script had an `offset` variable, which already exists in `Sprite`.
Both `offset` fields were deserialized to the value from the script, and I lost the previous value my sprite originally had.

![unexpected_shadowed_variable](https://cloud.githubusercontent.com/assets/1311555/16546343/aeb438b2-4147-11e6-821b-4c9c185d8dc9.png)

In the scene file you can see this:

```
[node name="arm" type="Sprite" parent="pivot"]

z/z = -1
texture = ExtResource( 2 )
centered = false
offset = Vector2( 0.1234, 0 ) <-----------------------------------
region = true
region_rect = Rect2( 0, 39, 89, 36 )
script/script = ExtResource( 3 )
color = Color( 1, 0, 0, 0.5 )
flicker_color = Color( 1, 0, 0, 0.3 )
flicker_interval_msec = 100
thickness = 3
offset = Vector2( 0.1234, 0 ) <-----------------------------------
```

Because you can inherit Node2D and still be able to put the script on Sprites, RigidBody or anything, it makes it hard to show a warning in the code editor.
So, we could show a popup when the script is added or when the scene is loaded? Or better, prepend something to the property so it doesn't interfer (`script/stuff`").
Any ideas?

**Comment by bojidar-bg** [comment]
Note that prepending something to the property the script creates would break compatibility with `set` and `get`, as well as all existing scenes

**Comment by vnen** [comment]
This should not be a problem anymore in 3.0. Is there anything to be done in the 2.1 branch?

**Comment by akien-mga** [comment]
If it's fixed in 3.0+, I think we can close. A fix for 2.1 would be welcome but it's easy to workaround this issue, so not high priority.
